vscode/microsoft-kiota/src/steps.ts

```diff
@@ -118,21 +119,21 @@ export async function searchSteps(searchCallBack: (searchQuery: string) => Thena
     return state;
 }
 
-export async function generateSteps(existingConfiguration: Partial<GenerateState>, languagesInformation?: LanguagesInformation, isDeeplinkEnabled?: boolean) {
+export async function generateSteps(existingConfiguration: Partial<GenerateState>, languagesInformation?: LanguagesInformation, deepLinkParams?: Partial<IntegrationParams>) {
     const state = { ...existingConfiguration } as Partial<GenerateState>;
     if (existingConfiguration.generationType && existingConfiguration.clientClassName && existingConfiguration.clientNamespaceName && existingConfiguration.outputPath && existingConfiguration.language &&
         typeof existingConfiguration.generationType === 'string' && existingConfiguration.clientNamespaceName === 'string' && typeof existingConfiguration.outputPath === 'string' && typeof existingConfiguration.language === 'string' &&
         existingConfiguration.generationType.length > 0 && existingConfiguration.clientClassName.length > 0 && existingConfiguration.clientNamespaceName.length > 0 && existingConfiguration.outputPath.length > 0 && existingConfiguration.language.length > 0) {
         return state;
     }
 
-    const isDeepLinkPluginNameProvided = isDeeplinkEnabled && state.pluginName;
-    const isDeepLinkGenerationTypeProvided = isDeeplinkEnabled && state.generationType;
-    const isDeepLinkPluginTypeProvided = isDeeplinkEnabled && state.pluginTypes;
-    const isDeepLinkLanguageProvided = isDeeplinkEnabled && state.language;
-    const isDeepLinkOutputPathProvided = isDeeplinkEnabled && state.outputPath;
-    const isDeepLinkClientClassNameProvided = isDeeplinkEnabled && state.clientClassName;
-
+    const deeplinkEnabled = !!deepLinkParams && isDeeplinkEnabled(deepLinkParams);
```

Comment:
This is why customizing everything for a single scenario e.g ttk is somewhat dirty. Now we are forced to pass deeplinkparams instead of a boolean which duplicates some of the details in existingConfiguration. 😢

Reply:
Yeah.   Though passing the object does help us fine tune the logic further down the line

Reply:
But we only use the source param in the deeplinkparams object. Everythin else is available from the transformed existing configuration. Isn't it easier to just pass the source as a param instead? Passing deeplinkparams which has somewhat similar values but different meaning in the generation type e.g a (generation) type of other in deeplinkparams, that would end up actually interpreted as an actual generationType=plugin of type OpenAI in existingconfig/state would like mislead a developer down the line making code maintenance harder.

Reply:
What you are saying is worth looking at further once we are done chasing GA. Especially since we are adding checks for source in multiple places. 

Putting in source as a parameter alone doesn't clarify what it's a source of. The transformation leaves the source of the deeplink out hence it makes sense for maintainer to know that it is a deeplink source. 

If we can look at stream lining this after GA, I'd be happy to since I have been aching for a bit of refactoring